### DISCUSSION TITLE
39SFxxx for flashrom adapter

### DISCUSSION THREAD

**Discussion by Chomemel** [sanni/cartreader#1030] [Need help]
Hi Sanni,
I saw you add support for [39F040 for the flashrom adapter](https://github.com/sanni/cartreader/wiki/39F040-PLCC32). Would it also work with a DIP version of the 39SF040 (with the right wiring of course)?

**Comment by sanni** [comment]
The PLCC 39SFxxx in the DIP adapter should have the same pinout as a real DIP 39SFxxx. So it should just work the same.
Maybe double-check the location of VCC and GND before using it just to be sure.

Note: The SNES maskrom has some pins in a different location but that gets "transformed" [here](https://github.com/sanni/cartreader/blob/master/Cart_Reader/FLASH.ino#L973). That's why we can use the SNES maskrom flash adapter. This "pin transformation" only gets applied when choosing the PLCC32 to SNES menu option.

And to answer your question about SNES repros with 39SFxxx, I added support for it just now. Untested but it should work, just use the SNES repro menu like you normally would and it should get detected.

**Comment by Chomemel** [comment]
Hi Sanni. Thanks for the answer. I'll give it a shot when I find some time.

For the the differents locations, I use a software that swap some bites in order redirect only 2 pins (OE and A16 or somthing like that)

If I understood right, if you choose PLCC32, the OSCR swap it, so you use an unmodified ROM, right? In my case, I'll use the 8bit flash way.

**Comment by Chomemel** [comment]
I finally had time to try. It works like a charm. I was able to erase and flash a new game.
![IMG_20250125_212458](https://github.com/user-attachments/assets/e7675f95-cf68-425d-83d8-0b4bfa144742)
![IMG_20250125_212634](https://github.com/user-attachments/assets/d6495496-d02f-42c2-bc2f-1dfcbd0cfd2d)

Thank for adding it.

**Comment by sanni** [comment]
Thanks for testing it. 😃
